api/client/node/next.ts

```diff
@@ -0,0 +1,77 @@
+import type { OpenAIStreamCallbacks } from 'ai'
+import type { TiktokenModel } from 'js-tiktoken'
+
+let encodingForModel: (model: TiktokenModel) => any | undefined
+
+type OpenAIUsage = {
+  total_tokens: number
+  prompt_tokens: number
+  completion_tokens: number
+}
+
+type OpenAIStreamCallbacksWithUsage = OpenAIStreamCallbacks & {
+  onUsage?: (usage: OpenAIUsage) => void
+}
+
+export function createOpenAIStreamCallback(
+  {
+    model,
+    prompts,
+  }: {
+    model: TiktokenModel
+    prompts: string[]
+  },
+  openAIStreamCallbacks: OpenAIStreamCallbacksWithUsage
+) {
+  // Tiktoken is an optional dependency, so we require it conditionally
+  if (!encodingForModel) {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { encodingForModel: encodingForModel_ } = require('js-tiktoken')
```

Comment:
`import` and `require` shouldn't be mixed

Reply:
ES6 imports are async by default. You can't do synchronous `import`, only with `require`. As this factory is synchronous, I don't see a way to use `import`. Btw I don't know any reason why mixing would impact a Node.js application.